azkaban-common/src/main/java/azkaban/storage/HdfsAuth.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2017 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ */
+
+package azkaban.storage;
+
+import azkaban.spi.AzkabanException;
+import azkaban.utils.Props;
+import com.google.inject.Inject;
+import java.io.IOException;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.security.UserGroupInformation;
+import org.apache.log4j.Logger;
+
+import static azkaban.Constants.ConfigurationKeys.*;
+import static java.util.Objects.*;
+
+
+/**
+ * This class helps in HDFS authorization and is a wrapper over Hadoop's {@link UserGroupInformation} class.
+ */
+public class HdfsAuth {
+  private static final Logger log = Logger.getLogger(HdfsAuth.class);
+
+  private final boolean isSecurityEnabled;
+
+  private UserGroupInformation loggedInUser;
```

Comment:
For our purposes yes. This is intended to be the user managing the azkaban app.
But we don't control what user and what keytab, therefore keeping it generic.